app/apollo/test/subscriptions.spec.js

```diff
@@ -497,6 +601,57 @@ describe('subscription graphql test suite', () => {
     }
   });
 
+  it('edit a subscription with custom attribute', async () => {
+    try {
+      //step1, edit the subscription
+      const result = await subscriptionApi.editSubscription(token77, {
+        orgId: org77._id,
+        uuid: subscription_04_uuid,
+        name: 'new-name',
+        groups:['new-tag'],
+        channelUuid: channel_04_uuid,
+        versionUuid: channelVersion_04_uuid,
+        custom: [
+          {
+            'key': 'forEnv',
+            'val': 'new'
+          },
+          {
+            'key': 'forType',
+            'val': 'new'
+          }
+        ],
+      });
+      const {
+        data: {
+          data: { editSubscription },
+        },
+      } = result;
+      expect(editSubscription.uuid).to.be.an('string');
+      //step2, get the updated subscription
+      const result2 = await subscriptionApi.subscription(token77, {
+        orgId: org77._id,
+        uuid: subscription_04_uuid,
+      });
+      const {
+        data: {
+          data: { subscription },
+        },
+      } = result2;
+      expect(subscription.name).to.equal('new-name');
+      expect(subscription.custom[0].val).to.equal('new');
+      expect(subscription.custom[1].val).to.equal('new');
+
+    } catch (error) {
+      if (error.response) {
+        console.error('error encountered:  ', error.response.data);
+      } else {
+        console.error('error encountered:  ', error);
+      }
+      throw error;
+    }
+  });
+
```

Comment:
All looks good. But if we begin to add three tests for each new GraphQL attribute then test code would be bloated reducing maintainability. Would it be possible to just add the attributes to the existing tests?